src/tools/genpolicy/rules.rego

```diff
@@ -636,56 +797,6 @@ allow_user(p_process, i_process) {
     # based on /etc/passwd and /etc/group from the container image.
 }
 
-allow_args(p_process, i_process, s_name) {
```

Comment:
todo:
- We need to keep supporting arg validation for old requests
-  But this change breaks old validation since we are no longer trying to expand variables in the args during policy generation

Reply:
Shifting this validation to be only done for old requests is still not supported. Because the args will still no longer be substituted in new policies, because we need the new env_map from the input (new agent) in order to validate them through substitution. Some options:
- Try to build the env_map from the input.Env array (in rego) and use new validation. This is a challenge to do this string manipulation correctly in rego 
- remove support for old requests. This would require new agent / genpolicy versions to be released in sync and would require customers to update both

Reply:
Put non substituted args on a new field in the policy and leave the old ones as they are currently (substituted). This will increase the size of the policy, but maintain compatibility